autogen/sr.rs

```diff
@@ -23,42 +23,131 @@ struct OperandTokens {
     name: Ident,
     /// Rust type used in structured representation.
     quantified_type: TokenStream,
+    /// Rust expression that initializes the structured value
+    /// based on the data repreentation.
+    lift_expression: TokenStream,
 }
 
 impl OperandTokens {
-    fn new(operand: &structs::Operand) -> Self {
+    fn new(operand: &structs::Operand, iter: &Ident) -> Self {
         let name = get_param_name(operand);
 
-        let ty = match operand.kind.as_str() {
+        let (ty, lift_value) = match operand.kind.as_str() {
             "IdRef" => match operand.name.trim_matches('\'') {
-                "Length" => quote! { Token<Constant> },
-                "Field Types" => quote! { StructMember },
-                "Parameter Types" => quote! { Token<Type> },
-                name if name.ends_with(" Type") => quote! { Token<Type> },
-                _ => quote! { spirv::Word },
+                "Length" => (
+                    quote! { Token<Constant> },
+                    quote! { self.constants.lookup(*value).unwrap() },
+                ),
+                "Field Types" => (
+                    quote! { StructMember },
+                    quote! { super::types::StructMember::new(value.clone()) },
+                ),
+                "Parameter Types" => (
```

Comment:
Yes. As I described in a separate comment, this `OpFunction` SR representation would be good to have so that we can use it internally. It doesn't need to be exposed to or used by the user.

Reply:
Okay then.